### DISCUSSION TITLE
[discussion] Connect browser-based client to a local browser-based server?

### DISCUSSION THREAD

**Discussion by willrogers007** [appy-one/acebase-client#10] [Ideas]
Not meant so much as an issue to be solved, but more of a discussion topic / tentative-feature-request-if-it-makes-sense.

In the docs I've seen an example of a client using a local server as a cache, but I haven't seen an example of a client connecting to a local server as it would to a remote server.

The client creation options seem to expect a host/port, whereas I'd love to just pass a living server object reference in.

I imagine this could have some wild uses, especially if plugging it into something like WebRTC. Could enable a peer-to-peer database system, piggybacking on the acebase-server's ability to have authentication, access rules & schemas.

Anyways, color me intrigued; I'd be excited to explore in that direction.

**Comment by appy-one** [comment]
Exactly what I've been thinking about the past weeks - using WebRTC for cross-browser synchronization. I'll be committing new IPC code for node clusters somewhere this week, which shares code for cross-browser-tab communication and leaves room for additional IPC implementations.
